java/dataflow-connector-examples/src/main/java/com/google/cloud/bigtable/dataflow/example/CsvImport.java

```diff
@@ -59,47 +66,124 @@
-        LOG.error("Failed to process input {}", c.element(), e);
+        LOG.error("Failed to process input {}", csvRowString, e);
         throw e;
       }
+    }
+
+    private Put makePutFromCsvRecord(CSVRecord record) {
+      Put rowPut = new Put(record.get(rowKeyColumnName).getBytes(StandardCharsets.UTF_8));
+      long timestamp = System.currentTimeMillis();
+
+      record.toMap().forEach((csvHeader, value) -> {
+        if (csvHeader.equals(rowKeyColumnName)) {
+          return;
+        }
 
+        rowPut.addColumn(bigTableFamilyNameBytes, csvHeader.getBytes(StandardCharsets.UTF_8), timestamp, value.getBytes(StandardCharsets.UTF_8));
+
+      });
+
+      return rowPut;
     }
-  };
 
-  public static interface BigtableCsvOptions extends CloudBigtableOptions {
+    private CSVFormat makeCsvFormatWithHeaders() {
+      return csvFormat.builder().setHeader(csvHeaders).setSkipHeaderRecord(true).build();
+    }
+
+    public static Builder builder() {
+      return new Builder().setCsvFormat(CSVFormat.DEFAULT);
+    }
+
+    private static class Builder {
+      private String bigTableFamilyName;
+
+      private CSVFormat csvFormat;
+
+      private List<String> csvHeaders;
+
+      private String rowKeyColumnName;
+
+      public Builder setBigTableFamilyName(String bigTableFamilyName) {
+        this.bigTableFamilyName = bigTableFamilyName;
+        return this;
+      }
+
+      public Builder setCsvFormat(CSVFormat csvFormat) {
+        this.csvFormat = csvFormat;
+        return this;
+      }
+
+      public Builder setCsvFormat(CSVFormat.Predefined csvFormat) {
+        this.csvFormat = csvFormat.getFormat();
+        return this;
+      }
+
+      public Builder setCsvHeaders(List<String> csvHeaders) {
+        this.csvHeaders = csvHeaders;
+        return this;
+      }
+
+      public Builder setRowKeyColumnName(String rowKeyColumnName) {
+        this.rowKeyColumnName = rowKeyColumnName;
+        return this;
+      }
+
+      public MutationTransform build() {
+        return new MutationTransform(csvFormat, bigTableFamilyName, csvHeaders, rowKeyColumnName);
+      }
+    }
+  }
+
+  public interface BigtableCsvOptions extends CloudBigtableOptions {
 
     @Description("The headers for the CSV file.")
-    String getHeaders();
+    List<String> getHeaders();
+
+    void setHeaders(List<String> headers);
+
+    @Description("The column Name to use as RowKey")
+    String getRowKeyHeader();
+
+    void setRowKeyHeader(String rowKeyHeader);
+
+    @Description("The CSV Format as per CSVFormat class. Default: Default")
+    @Default.String("Default")
+    CSVFormat.Predefined getCsvFormat();
+
+    void setCsvFormat(CSVFormat.Predefined predefinedCsvFormat);
+
+    @Description("The name of the BigTable column family to put the csv columns under")
+    @Default.String("csv")
+    String getBigTableColumnFamilyName();
 
-    void setHeaders(String headers);
+    void setBigTableColumnFamilyName(String bigTableColumnFamilyName);
```

Comment:
Bigtable is one word, so don't camelcase table

Reply:
Done